src/test/java/com/aws/greengrass/shadowmanager/sync/model/LocalUpdateSyncRequestTest.java

```diff
@@ -91,7 +92,7 @@ class LocalUpdateSyncRequestTest {
     @BeforeEach
     void setup() throws IOException {
         syncContext = new SyncContext(mockDao, mockUpdateThingShadowRequestHandler,
-                mock(DeleteThingShadowRequestHandler.class), mock(IotDataPlaneClientWrapper.class));
+                mock(DeleteThingShadowRequestHandler.class), mock(IotDataPlaneClientWrapper.class), new ShadowWriteSynchronizeHelper());
```

Comment:
Why not use a mock?

Reply:
since its just a map of locks, wouldn't mocking be more complicated to achieve the same functionality?

Reply:
wanted to exercise locking within sync testing which is why using real class here and cloud update test, while using mocks everywhere else